R/opts.R

```diff
@@ -917,6 +917,299 @@ opt_table_font <- function(
+    color = "blue"
+) {
+
+  if (!(length(style) == 1 && style %in% 1:6)) {
+    cli::cli_abort("The `style` value must be a number in the range of 1 to 6.")
+  }
+
+  if (!(length(color) == 1 && color %in% names(preset_colors))) {
+    cli::cli_abort(c(
+      "The `color` value must be one of seven specific colors:",
+      "*" = "\"blue\", \"orange\", \"aqua\", \"pink\", \"green\", \"red\", or \"gray\""
+    ))
+  }
+
+  # Get the set of parameters based on the `style` and `color` values
+  params <-
+    get_colorized_params(
+      styles_colors_params = styles_colors_params,
+      style = style,
+      color = color
+    )
+
+  tbl_colorized <-
+    tab_options(
+      dt_options_init(data = data),
+      table.border.top.color = params$table_hlines_color,
+      table.border.bottom.color = params$table_hlines_color,
+      heading.border.bottom.color = params$location_hlines_color,
+      column_labels.border.top.color = params$location_hlines_color,
+      column_labels.border.bottom.color = params$location_hlines_color,
+      row_group.border.top.color = params$location_hlines_color,
+      row_group.border.bottom.color = params$location_hlines_color,
+      summary_row.border.color = params$location_hlines_color,
+      grand_summary_row.border.color = params$location_hlines_color,
+      column_labels.background.color = params$column_labels_background_color,
+      stub.background.color = params$stub_background_color,
+      stub.border.style = params$stub_border_style,
+      stub.border.color = params$stub_border_color,
+      table_body.border.top.color = params$location_hlines_color,
+      table_body.border.bottom.color = params$location_hlines_color,
+      table_body.hlines.style = params$data_hlines_style,
+      table_body.hlines.color = params$data_hlines_color,
+      table_body.vlines.style = params$data_vlines_style,
+      table_body.vlines.color = params$data_vlines_color,
+      summary_row.background.color = params$summary_row_background_color,
+      grand_summary_row.background.color = params$grand_summary_row_background_color,
+      row.striping.background_color = params$row_striping_background_color
+    )
+
+  if (!is.na(params$table_outline_color)) {
+
+    tbl_colorized <-
+      opt_table_outline(
+        tbl_colorized,
+        width = px(3),
+        color = params$table_outline_color
+      )
+  }
+
+  tbl_colorized
+}
+
+# Colors for the `preset_colorized()` function
+preset_colors <-
+  list(
+    gray =    c("#F4F4F4", "#D5D5D5", "#929292", "#5F5F5F", "#000000"),
+    blue =    c("#EDF7FC", "#89D3FE", "#00A1FF", "#0076BA", "#004D80"),
+    orange =  c("#FFFFDB", "#FEFC66", "#FAE231", "#F8BA00", "#FF9400"),
+    aqua =    c("#EBFBF9", "#A5FEF2", "#14E6CF", "#01A99D", "#007C77"),
+    pink =    c("#FCF2F7", "#FF8CC6", "#EF5FA7", "#CB2A7B", "#99195F"),
+    green =   c("#EDF6E8", "#9AFD69", "#60D937", "#1EB100", "#027101"),
+    red =     c("#FEEDEC", "#FF968D", "#FF644E", "#ED220D", "#B51800")
+  )
```

Comment:
Would it make sense to document where these came from?

Reply:
I will document this in the `X09-styles_colors_params.R` script (in `data-raw`). Basically I chose some base colors, made one darker variation of each and then three lighter variations. This was in Sketch for the most part:

<img width="1436" alt="color_variations_sketch" src="https://user-images.githubusercontent.com/5612024/179257703-d331c6f1-af10-45b6-8f8d-b4ee6e99192c.png">

I'll put in some notes w.r.t. the color metrics for each (lightness, contrast with black/white text, etc.) in that file soon.

Reply:
Also, I changed the color name `aqua` to `cyan`. It was closer to `aquamarine` than `aqua` but `cyan` is even closer.